lib/RegisterAll.cpp

```diff
@@ -37,10 +37,15 @@ void mlir::tt::registerAllDialects(mlir::DialectRegistry &registry) {
 }
 
 void mlir::tt::registerAllPasses() {
-  // Register all dialect conversion passes
-  //
+  // Register all dialect conversion passes.
   mlir::tt::registerTTMLIRConversionPasses();
 
+  // Registering -remove-dead-values built-in mlir pass to optimize out the
+  // unused OPs/operands after conversion.
+  mlir::registerPass([]() -> std::unique_ptr<::mlir::Pass> {
+    return mlir::createRemoveDeadValuesPass();
+  });
+
```

Comment:
For the context, the way `stablehlo.reduce` works is that it is a generic OP which they combine with binary OPs, and it accepts initial value as an argument. For example to do what `ttnn.sum` does, they setup a constant initial value of 0 and then apply reduce with add on tensor by doing `add(initValue, tensor[0])`, then `add(result, tensor[1])`, etc. This is how it looks like:
```
%cst_2 = stablehlo.constant dense<0.000000e+00> : tensor<f32>
%21 = stablehlo.reduce(%20 init: %cst_2) applies stablehlo.add across dimensions = [1] : (tensor<128x10xf32>, tensor<f32>) -> tensor<128xf32>
```

Since TTIR/TTNN SumOp doesn't work with initial value, after we convert `stablehlo.reduce` to `ttir.sum` we end up with unused `stablehlo.constant` OP call. That is where this pass becomes useful, and I think it is also useful in general to have it in our pipeline.

Reply:
Can this be written?
```c++
mlir::registerPass(mlir::createRemoveDeadValuesPass);
```

Reply:
Thanks, that works! Not sure why autogenerated pass register functions are always generated with lambda, I copied from there.